lib/node_modules/@stdlib/blas/base/zdscal/src/addon.c

```diff
@@ -0,0 +1,65 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/blas/base/zdscal.h"
+#include "stdlib/blas/base/shared.h"
+#include "stdlib/napi/export.h"
+#include "stdlib/napi/argv.h"
+#include "stdlib/napi/argv_double.h"
+#include "stdlib/napi/argv_int64.h"
+#include "stdlib/napi/argv_strided_complex128array.h"
+#include <node_api.h>
+
+/**
+* Receives JavaScript callback invocation data.
+*
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 4 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
+	STDLIB_NAPI_ARGV_DOUBLE( env, alpha, argv, 1 );
+	STDLIB_NAPI_ARGV_STRIDED_COMPLEX128ARRAY( env, X, N, strideX, argv, 2 );
+	stdlib_complex128_t *x = (stdlib_complex128_t *)X;
+	API_SUFFIX(c_zdscal)( N, alpha, x, strideX );
```

Comment:
This isn't correct. You should be doing

```suggestion
	API_SUFFIX(c_zdscal)( N, alpha, (void *)X, strideX );
```

Again, why are you deviating from existing prior art such as https://github.com/stdlib-js/stdlib/blob/bab1647c1309043335a7ec9badb8f3e629bf9136/lib/node_modules/%40stdlib/blas/base/ccopy/src/addon.c#L41? You need to explain your decision-making in your OP.

Reply:
It is apparent that you are lacking context. Why `void *`? That is because we follow CBLAS. As such, any user familiar with BLAS who attempts to use our C implementations will be expecting that they should perform a `void *)` cast. Us using a custom stdlib type doesn't match that expectation. There is historical reasons for using `void *`, namely platforms did not always have a standardized complex number representation.